### PR TITLE
Cleaner `force_merge` signature

### PR DESCRIPTION
**Pull request by pascutto**
Add a way to get the witness from the index so that it's not required in the `force_merge` signature.
`get_witness` is currently not very nice so review is appreciated.

### CODE REVIEW

File: src/index.ml [hunk 1]
```diff
@@ -422,9 +422,23 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
         IO.set_generation t.log generation;
         t.generation <- generation
 
-  let force_merge t key value =
+  exception Found of entry
```
Comment (CraigFe) [inline src/index.ml]: Consider defining this exception in `get_witness` with
```ocaml
  let get_witness t =
    let exception Found of entry in
```
to limit the damage.

File: src/index.ml [hunk 2]
```diff
@@ -422,9 +422,22 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
...
+    try
+      Tbl.iter (fun _ entry -> raise (Found entry)) t.log_mem;
+      match t.index with
+      | None -> None
+      | Some index ->
+          let buf = Bytes.create entry_size in
+          let n = IO.read index.io ~off:0L buf in
+          assert (n = entry_size);
+          Some (decode_entry buf 0)
+    with Found e -> Some e
```
Comment (Julow) [inline src/index.ml]: You could `try` only the `Tbl.iter ` line with `match Tbl.iter .. with exception Found -> ... | () -> ...` to avoid having unrelated code in the `try` block.
